internal/gcsx/connection.go

```diff
@@ -36,9 +36,17 @@ func NewConnection(cfg *gcs.ConnConfig) (c *Connection, err error) {
 		return
 	}
 
+	var auth option.ClientOption
+	if cfg.TokenSource == nil {
```

Comment:
When the endpoint is overridden to a non-google url, the TokenSource would be 
```
oauth2.StaticTokenSource(&oauth2.Token{})
```
according to `getConn`. Why do we need to handle `cfg.TokenSource == nil` which never happens?

Reply:
Removed.  I tested this against GCS and fake-gcs-server to make sure that both still work.